src/tools/mixed.hh

```diff
@@ -0,0 +1,221 @@
+/*
+ * mixed.hh
+ *
+ *  Created on: Apr 8, 2019
+ *      Author: jb
+ */
+
+#ifndef SRC_TOOLS_MIXED_HH_
+#define SRC_TOOLS_MIXED_HH_
+
+#include <tuple>
+#include <memory>
+#include <iostream>
+#include <type_traits>
+
+const int __spacedim = 3;
+
+using Dim = unsigned int;
+
+template<template<Dim ...> class T>
+using _MixedBase = std::tuple<T<0>, T<1>, T<2>, T<3>>;
+
+template< template<Dim ...> class T>
+class Mixed : public _MixedBase<T> {
+/**
+ * Template to simplify storing and passing tuples of instances of dimension parametrized templates.
+ * Currently instances for dim = 0,1,2,3 are created. We assume spacedim=3.
+ * Usage:
+ *
+ * fe_order = 10;
+ * auto gauss = Mixed<QGauss>(q_order);      // template <int dim> class QGauss;
+ * auto mapping = MixedSpaceDim<Mapping>();  // template <int dim, int spacedim> class Mapping;
+ * ...
+ * fe_values = FEValues<dim>(mapping.get<dim>(), gauss.get<dim>(), ...)
+ *
+ * All parameters of the Mixed<T>(...) constructor are forwarder (by value) to the T(...) constructor for every dimension.
+ * Only forwarding by value is supported due to problems with MixedPtr.
+ * Can not resolve copy constructor correctly.
+ *
+ */
+public:
+    template < template<Dim...> class TT>
+    static Mixed<T> cast_to_parent_template(const Mixed<TT> &other)
+    {
+    	//bool parent = std::is_base_of<T<0>, TT<0>>::value;
+        return Mixed<T>(
+        		T<0>(other.get<0>()),
+				T<1>(other.get<1>()),
+				T<2>(other.get<2>()),
+				T<3>(other.get<3>()));
+    }
+
+    template < template<Dim...> class TT>
+    Mixed(const Mixed<TT> &other)
+    : _MixedBase<T>(
+            other.get<0>(),
```

Comment:
Possibly try explicit call of the copy constructor here:
`T<0>(other.get<0>())` as in the cast method